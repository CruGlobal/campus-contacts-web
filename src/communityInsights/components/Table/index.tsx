import React, { useState } from 'react';
import { useQuery } from 'react-apollo-hooks';
import { useTranslation } from 'react-i18next';
import styled from '@emotion/styled';
import { withTheme } from 'emotion-theming';

import leftArrow from '../../assets/icons/arrow-left.svg';
import leftArrowActive from '../../assets/icons/arrow-left-active.svg';
import rightArrow from '../../assets/icons/arrow-right.svg';
import rightArrowActive from '../../assets/icons/arrow-right-active.svg';

const LoadingContainer = styled.div`
    text-align: center;
    padding: 25px;
`;

const TableContainer = styled.table`
    table-layout: fixed;
    width: 100%;
`;

const Header = styled.thead`
    text-align: left;
`;

const Content = styled.tbody``;

const Column = styled.th`
    font-weight: normal;
    font-size: 14px;
    line-height: 60px;
    color: ${({ theme }) => theme.colors.secondary};
    padding: 0 15px;
`;

const Row = styled.tr`
    text-align: left;
    height: 64px;

    :nth-of-type(odd) {
        background-color: #eceef2;
    }
`;

const Cell = styled.td`
    font-weight: normal;
    font-size: 14px;
    line-height: 20px;
    color: ${({ theme }) => theme.colors.primary};
    padding: 0 15px;
    max-width: 200px;
    overflow: hidden;
    text-overflow: ellipsis;
    white-space: nowrap;
    &:first-of-type {
        padding: 0 25px;
    }
    &:last-child {
        padding: 0 5px;
    }
`;

const Pagination = styled.div`
    width: 100%;
    height: 64px;
    display: flex;
    align-items: center;
    justify-content: center;
`;

const LeftArrow = styled.div`
    height: 16px;
    width: 16px;
    background-size: contain;
    background-repeat: no-repeat;
    background-image: url(${leftArrow});
    margin-right: 20px;

    &.active {
        cursor: pointer;
        background-image: url(${leftArrowActive});
    }
`;

const RightArrow = styled.div`
    height: 16px;
    width: 16px;
    background-size: contain;
    background-repeat: no-repeat;
    background-image: url(${rightArrow});
    margin-left: 20px;

    &.active {
        cursor: pointer;
        background-image: url(${rightArrowActive});
    }
`;

const Pages = styled.div`
    display: flex;
`;
const Page = styled.div`
    width: 32px;
    height: 32px;
    font-size: 14px;
    line-height: 20px;
    color: ${({ theme }) => theme.colors.secondary};
    align-items: center;
    display: flex;
    justify-content: center;

    &:not(:last-child) {
        margin-right: 4px;
    }

    &.active {
        border-radius: 16px;
        background-color: ${({ theme }) => theme.colors.highlight};
        color: #ffffff;
    }
`;

const PAGE_SIZE = 5;
const PAGE_NUMBERS_SIZE = 5;

interface Props {
    query: any;
    headers: any;
    mapRows: (data: any) => any;
    mapPage: (data: any) => any;
    variables: any;
}

const Table = ({ query, headers, mapRows, mapPage, variables }: Props) => {
    const [pageNumber, setPageNumber] = useState(1);

    const { data, loading, refetch } = useQuery(query, {
        variables: {
            ...variables,
            first: PAGE_SIZE,
        },
    });
    const { t } = useTranslation('insights');

    if (loading) {
        return <LoadingContainer>{t('loading')}</LoadingContainer>;
    }

    const rows = mapRows(data);
    const page = mapPage(data);

    const nextPage = () => {
        if (page.hasNextPage) {
            refetch({
                first: PAGE_SIZE,
                last: undefined,
                after: page.endCursor,
                before: undefined,
            });
            setPageNumber(pageNumber + 1);
        }
    };

    const previousPage = () => {
        if (page.hasPreviousPage) {
            refetch({
                first: undefined,
                last: PAGE_SIZE,
                after: undefined,
                before: page.startCursor,
            });
            setPageNumber(pageNumber > 1 ? pageNumber - 1 : 0);
        }
    };

    const renderPages = () => {
        const pages = [];
        const pageIndex = Math.floor(PAGE_NUMBERS_SIZE / 2);
        const startIndex =
            pageNumber - pageIndex > 0 ? pageNumber - pageIndex : 1;
        for (let i = startIndex; i < startIndex + PAGE_NUMBERS_SIZE; i++) {
            pages.push(
                <Page key={i} className={i === pageNumber ? 'active' : ''}>
                    {i}
                </Page>,
            );
        }
        return pages;
    };

    return (
        <div>
            <TableContainer>
                <Header>
                    <tr>
                        {headers.map((header: string, index: string) => (
                            <Column key={index}>{header}</Column>
                        ))}
                    </tr>
                </Header>
                <Content>
                    {rows.map((row: any, index: string) => (
                        <Row key={index}>
                            {row.map((cell: string, index: string) => (
                                <Cell key={index}>{cell}</Cell>
                            ))}
                        </Row>
                    ))}
                </Content>
            </TableContainer>
            <Pagination>
                <LeftArrow
                    className={page.hasPreviousPage ? 'active' : ''}
                    onClick={previousPage}
                />
                <Pages>{renderPages()}</Pages>
                <RightArrow
                    className={page.hasNextPage ? 'active' : ''}
                    onClick={nextPage}
                />
            </Pagination>
        </div>
    );
};

export default withTheme(Table);
