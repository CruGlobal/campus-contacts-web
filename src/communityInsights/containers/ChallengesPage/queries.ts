import gql from 'graphql-tag';

const GET_IMPACT_CHALLENGES = gql`
    query impactReportStepsCount($id: ID!) {
        community(id: $id) {
            impactReport {
                stepsCount
            }
        }
    }
`;

const GET_CHALLENGES = gql`
    query communityChallenges(
        $id: ID!
        $first: Int
        $last: Int
        $after: String
        $before: String
        $sortBy: [CommunityChallengesSortEnum!]
    ) {
        community(id: $id) {
            communityChallenges(
                first: $first
                last: $last
                after: $after
                before: $before
                sortBy: $sortBy
            ) {
                nodes {
                    id
                    title
                    acceptedCount
                    completedCount
                    createdAt
                    endDate
                }
                pageInfo {
                    hasNextPage
                    startCursor
                    endCursor
                    hasPreviousPage
                }
            }
        }
    }
`;

export { GET_CHALLENGES, GET_IMPACT_CHALLENGES };
