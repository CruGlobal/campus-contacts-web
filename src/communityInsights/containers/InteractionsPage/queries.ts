import gql from 'graphql-tag';

const INTERACTIONS_TOTAL_REPORT = gql`
    query impactReportInteractionsCount($id: ID!) {
        community(id: $id) {
            impactReport {
                interactionsCount
                interactionsReceiversCount
            }
        }
    }
`;

const INTERACTIONS_TOTAL_COMPLETED_REPORT = gql`
    query communityReportInteractions(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                interactions {
                    nodes {
                        interactionCount
                        interactionType {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const INTERACTIONS_COMPLETED_REPORT = gql`
    query communityReportDaysInteractions(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        interactionsCount
                        interactionResults {
                            nodes {
                                count
                                interactionType {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

export {
    INTERACTIONS_TOTAL_REPORT,
    INTERACTIONS_TOTAL_COMPLETED_REPORT,
    INTERACTIONS_COMPLETED_REPORT,
};
