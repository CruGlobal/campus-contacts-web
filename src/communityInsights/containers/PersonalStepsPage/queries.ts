import gql from 'graphql-tag';

const GET_IMPACT_REPORT_MOVED = gql`
    query impactReportStageProgressionCount($id: ID!) {
        community(id: $id) {
            impactReport {
                stageProgressionCount
            }
        }
    }
`;

const GET_IMPACT_REPORT_STEPS_TAKEN = gql`
    query impactReportPersonalStepsCompletedCount($id: ID!) {
        community(id: $id) {
            impactReport {
                personalStepsCompletedCount
            }
        }
    }
`;

const GET_STAGES_REPORT_MEMBER_COUNT = gql`
    query communityReportStagesPersonalMemberCount(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        memberCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STAGES_REPORT_STEPS_ADDED = gql`
    query communityReportStagesPersonalStepsAdded(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        personalStepsAddedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityReportDaysPersonalSteps(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        personalStepsCompletedCount
                        stageResults {
                            nodes {
                                personalStepsCompletedCount
                                stage {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_SUMMARY = gql`
    query communityReportStagesPersonalStepsCompleted(
        $id: ID!
        $period: String!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        personalStepsCompletedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_MOVED,
    GET_IMPACT_REPORT_STEPS_TAKEN,
    GET_STAGES_REPORT_MEMBER_COUNT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT_STEPS_ADDED,
    GET_TOTAL_STEPS_COMPLETED_SUMMARY,
};
