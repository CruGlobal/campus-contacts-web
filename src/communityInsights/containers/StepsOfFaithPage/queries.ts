import gql from 'graphql-tag';

const GET_IMPACT_REPORT_TAKEN = gql`
    query impactReportOtherStepsCompleted($id: ID!) {
        community(id: $id) {
            impactReport {
                othersStepsCompletedCount
                othersStepsReceiversCompletedCount
            }
        }
    }
`;

const GET_TOTAL_STEPS_COMPLETED_REPORT = gql`
    query communityReportStagesOthersStepsCompleted(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        othersStepsCompletedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_STEPS_COMPLETED_REPORT = gql`
    query communityReportDaysOthersSteps(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                daysReport {
                    nodes {
                        date
                        othersStepsCompletedCount
                        stageResults {
                            nodes {
                                othersStepsCompletedCount
                                stage {
                                    name
                                }
                            }
                        }
                    }
                }
            }
        }
    }
`;

const GET_STAGES_REPORT = gql`
    query communityReportStagesOthersStepsAdded(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        othersStepsAddedCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

const GET_IMPACT_REPORT_REACHED = gql`
    query impactReportMembersStageProgressionCount($id: ID!) {
        community(id: $id) {
            impactReport {
                membersStageProgressionCount
            }
        }
    }
`;

const GET_STAGES_PEOPLE_REPORT = gql`
    query communityReportStagesOthersContactCount(
        $period: String!
        $id: ID!
        $endDate: ISO8601DateTime!
    ) {
        community(id: $id) {
            report(period: $period, endDate: $endDate) {
                stagesReport {
                    nodes {
                        contactCount
                        stage {
                            name
                        }
                    }
                }
            }
        }
    }
`;

export {
    GET_IMPACT_REPORT_TAKEN,
    GET_IMPACT_REPORT_REACHED,
    GET_TOTAL_STEPS_COMPLETED_REPORT,
    GET_STEPS_COMPLETED_REPORT,
    GET_STAGES_REPORT,
    GET_STAGES_PEOPLE_REPORT,
};
