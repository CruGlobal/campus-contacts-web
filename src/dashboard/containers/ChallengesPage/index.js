import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';

import Card from '../../components/Card';
import AppContext from '../../appContext';
import Table from '../../components/Table';
import ImpactInfo from '../../components/ImpactInfo';

import {
    GET_IMPACT_CHALLENGES,
    GET_CHALLENGES,
    GET_GLOBAL_CHALLENGES,
} from './queries';

const ChallengesPage = () => {
    const { t } = useTranslation('insights');
    const { orgId } = useContext(AppContext);

    const mapRows = data => {
        const percentage = (a, b) => {
            const value = b === 0 ? 0 : Math.floor((a / b) * 100);
            return `${a} (${value}%)`;
        };

        const duration = (start, end) => {
            const startDate = moment(start).startOf('day');
            const endDate = moment(end).startOf('day');
            const days = moment.duration(endDate.diff(startDate)).asDays();
            return days === 1 ? '1 Day' : `${days} Days`;
        };

        const range = (start, end) => {
            const format = 'MM/DD/YYYY';
            const startDate = moment(start).format(format);
            const endDate = moment(end).format(format);
            return `${startDate} - ${endDate}`;
        };

        const rows = [];
        data.globalCommunityChallenges.nodes.forEach(
            ({
                id,
                title,
                acceptedCount,
                completedCount,
                createdAt,
                endDate,
            }) => {
                const cells = [];
                cells.push(id);
                cells.push(title);
                cells.push(acceptedCount);
                cells.push(percentage(completedCount, acceptedCount));
                cells.push(duration(createdAt, endDate));
                cells.push(range(createdAt, endDate));
                rows.push(cells);
            },
        );
        return rows;
    };

    const mapPage = data => {
        return data.globalCommunityChallenges.pageInfo;
    };

    return (
        <div>
            <ImpactInfo
                query={GET_IMPACT_CHALLENGES}
                text={report =>
                    t('challenges.total', {
                        count:
                            report.communityReport.impactReport.challengesCount,
                    })
                }
                variables={{ organizationId: orgId }}
            />
            <Card
                title={t('challenges.title')}
                subtitle={t('challenges.subTitle')}
                noPadding={true}
            >
                <Table
                    query={GET_GLOBAL_CHALLENGES}
                    variables={{ id: orgId }}
                    headers={[
                        '',
                        '',
                        t('challenges.join'),
                        t('challenges.completed'),
                        t('challenges.length'),
                        t('challenges.dates'),
                    ]}
                    mapRows={mapRows}
                    mapPage={mapPage}
                />
            </Card>
        </div>
    );
};

export default ChallengesPage;

ChallengesPage.propTypes = {};
