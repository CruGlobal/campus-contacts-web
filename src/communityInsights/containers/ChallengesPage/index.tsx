import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';

import Card from '../../components/Card';
import { AppContext } from '../../../appContext';
import Table from '../../components/Table';
import ImpactInfo from '../../components/ImpactInfo';

import { GET_IMPACT_CHALLENGES, GET_CHALLENGES } from './queries';
// GET_IMPACT_CHALLENGES INTERFACES
import { impactReportStepsCount } from './__generated__/impactReportStepsCount';
// GET_CHALLENGES INTERFACES
import {
    communityChallenges,
    communityChallenges_community_communityChallenges_nodes as communityChallengesNodes,
} from './__generated__/communityChallenges';

const ChallengesPage = () => {
    const { t } = useTranslation('insights');
    const { orgId } = useContext(AppContext);

    const mapRows = (data: communityChallenges) => {
        const percentage = (a: number, b: number) => {
            const value = b === 0 ? 0 : Math.floor((a / b) * 100);
            return `${a} (${value}%)`;
        };

        const duration = (start: string, end: string) => {
            const startDate = moment(start).startOf('day');
            const endDate = moment(end).startOf('day');
            const days = Math.floor(
                moment.duration(endDate.diff(startDate)).asDays(),
            );
            return days === 1 ? '1 Day' : `${days} Days`;
        };

        const range = (start: string, end: string) => {
            const format = 'MM/DD/YYYY';
            const startDate = moment(start).format(format);
            const endDate = moment(end).format(format);
            return `${startDate} - ${endDate}`;
        };

        return data.community.communityChallenges.nodes.map(
            ({
                id,
                title,
                acceptedCount,
                completedCount,
                createdAt,
                endDate,
            }: communityChallengesNodes) => [
                id,
                title,
                acceptedCount,
                percentage(completedCount, acceptedCount),
                duration(createdAt, endDate),
                range(createdAt, endDate),
            ],
        );
    };

    const mapPage = (data: communityChallenges) => {
        return data.community.communityChallenges.pageInfo;
    };

    return (
        <div>
            <ImpactInfo
                query={GET_IMPACT_CHALLENGES}
                text={(report: impactReportStepsCount) =>
                    t('challenges.total', {
                        count: report.community.impactReport.stepsCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card
                title={t('challenges.title')}
                subtitle={t('challenges.subTitle')}
                noPadding={true}
                noMarginBottom={true}
            >
                <Table
                    query={GET_CHALLENGES}
                    variables={{
                        id: orgId,
                        sortBy: 'createdAt_DESC',
                    }}
                    headers={[
                        '',
                        '',
                        t('challenges.join'),
                        t('challenges.completed'),
                        t('challenges.length'),
                        t('challenges.dates'),
                    ]}
                    mapRows={mapRows}
                    mapPage={mapPage}
                />
            </Card>
        </div>
    );
};

export default ChallengesPage;
