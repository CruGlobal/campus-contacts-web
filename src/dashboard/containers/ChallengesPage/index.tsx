import React, { useContext } from 'react';
import { useTranslation } from 'react-i18next';
import moment from 'moment';

import Card from '../../components/Card';
import AppContext from '../../appContext';
import Table from '../../components/Table';
import ImpactInfo from '../../components/ImpactInfo';

import { GET_IMPACT_CHALLENGES, GET_CHALLENGES } from './queries';

const ChallengesPage = () => {
    const { t } = useTranslation('insights');
    const { orgId }: any = useContext(AppContext);

    const mapRows = (data: any) => {
        const percentage = (a: number, b: number) => {
            const value = b === 0 ? 0 : Math.floor((a / b) * 100);
            return `${a} (${value}%)`;
        };

        const duration = (start: Date, end: Date) => {
            const startDate = moment(start).startOf('day');
            const endDate = moment(end).startOf('day');
            const days = Math.floor(
                moment.duration(endDate.diff(startDate)).asDays(),
            );
            return days === 1 ? '1 Day' : `${days} Days`;
        };

        const range = (start: Date, end: Date) => {
            const format = 'MM/DD/YYYY';
            const startDate = moment(start).format(format);
            const endDate = moment(end).format(format);
            return `${startDate} - ${endDate}`;
        };

        return data.community.communityChallenges.nodes.map(
            ({
                id,
                title,
                acceptedCount,
                completedCount,
                createdAt,
                endDate,
            }: any) => [
                id,
                title,
                acceptedCount,
                percentage(completedCount, acceptedCount),
                duration(createdAt, endDate),
                range(createdAt, endDate),
            ],
        );
    };

    const mapPage = (data: any) => {
        return data.community.communityChallenges.pageInfo;
    };

    return (
        <div>
            <ImpactInfo
                query={GET_IMPACT_CHALLENGES}
                text={report =>
                    t('challenges.total', {
                        count: report.community.impactReport.interactionsCount,
                    })
                }
                variables={{ id: orgId }}
            />
            <Card
                title={t('challenges.title')}
                subtitle={t('challenges.subTitle')}
                noPadding={true}
            >
                <Table
                    query={GET_CHALLENGES}
                    variables={{
                        id: orgId,
                        sortBy: 'createdAt_DESC',
                    }}
                    headers={[
                        '',
                        '',
                        t('challenges.join'),
                        t('challenges.completed'),
                        t('challenges.length'),
                        t('challenges.dates'),
                    ]}
                    mapRows={mapRows}
                    mapPage={mapPage}
                />
            </Card>
        </div>
    );
};

export default ChallengesPage;

ChallengesPage.propTypes = {};
