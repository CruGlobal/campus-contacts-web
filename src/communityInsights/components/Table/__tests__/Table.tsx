import React from 'react';
import gql from 'graphql-tag';
import { waitForElement } from '@testing-library/react';

import { renderWithContext } from '../../../../testUtils';
import { GET_CHALLENGES } from '../../../containers/ChallengesPage/queries';

import Table from '..';

describe('<Table />', () => {
    it('should render properly in loading', async () => {
        const mapRows = () => [['cell1', 'cell2'], ['cell3', 'cell4']];
        const mapPage = () => ({});

        const { snapshot, unmount } = renderWithContext(
            <Table
                nullContent={'challengesCompleted'}
                query={GET_CHALLENGES}
                headers={['header-1', 'header-2', 'header-3']}
                mapRows={mapRows}
                mapPage={mapPage}
                variables={{}}
            />,
        );
        snapshot();
        unmount();
    });

    it('should render properly with data', async () => {
        const mapRows = () => [['cell1', 'cell2'], ['cell3', 'cell4']];
        const mapPage = () => ({});

        const { snapshot, getByText } = renderWithContext(
            <Table
                nullContent={'challengesCompleted'}
                query={GET_CHALLENGES}
                headers={['header-1', 'header-2', 'header-3']}
                mapRows={mapRows}
                mapPage={mapPage}
                variables={{}}
            />,
            {
                mocks: {
                    Query: () => ({
                        globalCommunityChallenges: () => ({
                            nodes: () => [],
                            pageInfo: () => {},
                        }),
                    }),
                },
                appContext: {
                    orgId: '1',
                },
            },
        );

        await waitForElement(() => getByText('header-1'));
        snapshot();
    });
});
