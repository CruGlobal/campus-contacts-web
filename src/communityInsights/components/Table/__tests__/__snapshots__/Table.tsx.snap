// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render properly in loading 1`] = `
.emotion-1 {
  width: 100%;
  height: 190px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #ffffff;
  border-radius: 8px;
}

.emotion-0 {
  text-align: center;
  width: 720px;
  color: #007398;
  font-size: 16px;
  font-weight: 300;
  line-height: 24px;
  margin: 10px 0;
}

<div>
  <div
    class="emotion-1"
  >
    <img
      src="mockFile"
    />
    <p
      class="emotion-0"
    >
      This community does not have any challenges. Open the MissionHub mobile app and create a community challenge.
    </p>
  </div>
</div>
`;

exports[`<Table /> should render properly with data 1`] = `
.emotion-11 {
  width: 100%;
}

.emotion-3 {
  text-align: left;
}

.emotion-0 {
  font-weight: normal;
  font-size: 14px;
  line-height: 60px;
  color: #B4B6BA;
  padding: 0 15px;
}

.emotion-6 {
  text-align: left;
  height: 64px;
}

.emotion-6:nth-child(odd) {
  background-color: #eceef2;
}

.emotion-4 {
  font-weight: normal;
  font-size: 14px;
  line-height: 20px;
  color: #505256;
  padding: 0 15px;
  max-width: 200px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
}

.emotion-20 {
  width: 100%;
  height: 64px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-12 {
  height: 16px;
  width: 16px;
  background-size: contain;
  background-repeat: no-repeat;
  background-image: url(mockFile);
  margin-right: 20px;
}

.emotion-12.active {
  cursor: pointer;
  background-image: url(mockFile);
}

.emotion-18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-13 {
  width: 32px;
  height: 32px;
  font-size: 14px;
  line-height: 20px;
  color: #B4B6BA;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-13:not(:last-child) {
  margin-right: 4px;
}

.emotion-13.active {
  border-radius: 16px;
  background-color: #3CC8E6;
  color: #ffffff;
}

.emotion-19 {
  height: 16px;
  width: 16px;
  background-size: contain;
  background-repeat: no-repeat;
  background-image: url(mockFile);
  margin-left: 20px;
}

.emotion-19.active {
  cursor: pointer;
  background-image: url(mockFile);
}

<div>
  <div>
    <table
      class="emotion-11"
    >
      <thead
        class="emotion-3"
      >
        <tr>
          <th
            class="emotion-0"
          >
            header-1
          </th>
          <th
            class="emotion-0"
          >
            header-2
          </th>
          <th
            class="emotion-0"
          >
            header-3
          </th>
        </tr>
      </thead>
      <tbody
        class="emotion-10"
      >
        <tr
          class="emotion-6"
        >
          <td
            class="emotion-4"
          >
            cell1
          </td>
          <td
            class="emotion-4"
          >
            cell2
          </td>
        </tr>
        <tr
          class="emotion-6"
        >
          <td
            class="emotion-4"
          >
            cell3
          </td>
          <td
            class="emotion-4"
          >
            cell4
          </td>
        </tr>
      </tbody>
    </table>
    <div
      class="emotion-20"
    >
      <div
        class=" emotion-12"
      />
      <div
        class="emotion-18"
      >
        <div
          class="active emotion-13"
        >
          1
        </div>
        <div
          class=" emotion-13"
        >
          2
        </div>
        <div
          class=" emotion-13"
        >
          3
        </div>
        <div
          class=" emotion-13"
        >
          4
        </div>
        <div
          class=" emotion-13"
        >
          5
        </div>
      </div>
      <div
        class=" emotion-19"
      />
    </div>
  </div>
</div>
`;
